doc/adding_new_cmd.md

```diff
@@ -0,0 +1,114 @@
+# Adding a new cli command
+
+
+The mobile CLI uses the [cobra](https://github.com/spf13/cobra) library to provide a consistent framework for building out the CLI tool.
+
+
+## Adding a new base command
+### What is a base command
+A base command is a command that works with an entirely new resource or that adds a new verb i.e (get, delete)
+### Changes required to add a new base command
+to add a new base command that operates on a new resource type you should add a new file under the ```pkg/cmd``` directory
+named ```<resourceName>.go``` with an accompanying test file ```<resourceName>_test.go```
+Once this is done, then you should follow the patterns in the other base commands:
+
+- Create a new type  ```<resourceName>Cmd```
+- Create new constructor ```New<resourceName>Cmd``` 
+- Dependencies such as the kubernetes clients should be passed to this constructor as their interface types to allow for simpler testing
+- The cobra commands should then be made methods from this the base type:
```

Comment:
I'm not sure what this sentence means, could it be: `The cobra commands should then be implemented as methods`?

Reply:
updated to say ``` The cobra commands should then be returned from methods from this the base type. See below```